src/wasm/wasm-s-parser.cpp

```diff
@@ -1302,6 +1302,9 @@ static size_t parseMemAttributes(Element& s,
   align = fallbackAlign;
   while (!s[i]->isList()) {
     const char* str = s[i]->c_str();
+    if (strncmp(str, "align", 5) != 0 && strncmp(str, "offset", 6) != 0) {
```

Comment:
I'm confused, what does this do?

Reply:
These instructions are the first to have a memarg (align and offset) followed by another immediate argument. This parsing routine previously expected the align and offset to be the only non-expression operands (see the loop condition), so it wasn't able to parse the new instructions correctly. This addition makes the loop bail out instead of raise an error when it encounters a non-memarg immediate argument.

Reply:
I see, thanks. Maybe a comment on the function (saying what it parses, and that it bails on anything else)?